haskell_scotty/ClientRepo.hs

```diff
@@ -16,3 +18,19 @@ getClients = do
 
 instance QueryResults Client where
   convertResults fs vs = Client $... zip fs vs
+
+saveClient :: Client -> IO Client
+saveClient client = do
+  conn <- connectDB
+  execute conn "INSERT INTO clients \
+    \(name,address,city,state,zip,country,email,contact,phone,created_at,updated_at) values \
+    \(?,?,?,?,?,?,?,?,?,?,?)" client
+  results <- query_ conn "SELECT LAST_INSERT_ID()"
+  let [Only clientId] = results
+  return client { Client.id = clientId }
+
+instance QueryParams Client where
+  renderParams
+    (Client _ name address city state zip country email contact phone created_at updated_at) =
+      [render name, render address, render city, render state, render zip, render country, render email, render contact, render phone, render created_at, render updated_at]
```

Comment:
Oh, I see, because `map :: (a -> b) -> [a] -> [b]`. Even though a list can have heterogeneous types, the `map` function must operate on a list of elements of the same type. Wah wah. There's nothing like a we-don't-care-as-long-as-it-can-be-`render`ed  type?